SimPEG/regularization/base.py

```diff
@@ -28,98 +26,266 @@ class BaseRegularization(BaseObjectiveFunction):
 
     """
 
-    def __init__(self, mesh=None, **kwargs):
-        super(BaseRegularization, self).__init__()
-        self.regmesh = RegularizationMesh(mesh)
-        if "indActive" in kwargs.keys():
-            indActive = kwargs.pop("indActive")
-            self.regmesh.indActive = indActive
-        utils.setKwargs(self, **kwargs)
+    _model = None
+
+    def __init__(
+        self,
+        mesh: RegularizationMesh | BaseMesh,
+        active_cells: np.ndarray | None = None,
+        mapping: maps.IdentityMap | None = None,
+        reference_model: np.ndarray | None = None,
+        units: str | None = None,
```

Comment:
From what I understood the `units` arguments is new. Might worth adding a description for it on the docstring of the class. I think it currently only does something if `units="radians"`. If we don't add that to the docstring is difficult for users to get and idea of what strings can be passes without digging into the code.